Assets/Scripts/UI/Mouse/MouseCursor.cs

```diff
@@ -0,0 +1,283 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+namespace ProjectPorcupine.Mouse
+{
+    public class MouseCursor
+    {
+        /// <summary>
+        /// The default color for the text.
+        /// </summary>
+        public static readonly Color TextColor = Color.white;
+
+        /// <summary>
+        /// The color of the background.
+        /// </summary>
+        public static readonly Color BackgroundColor = new Color(7f / 255f, 69f / 255f, 87f / 255f, 1f);
+
+        /// <summary>
+        /// The color of the outline of the text.
+        /// </summary>
+        public static readonly Color TextOutlineColor = new Color(7f / 255f, 70f / 255f, 92f / 255f, 1f);
+
+        /// <summary>
+        /// The color of the outline of the text.
+        /// </summary>
+        public static readonly Color BackgroundOutlineColor = new Color(206f / 255f, 242f / 255f, 242f / 255f, 1f);
+
+        private Texture2D cursorTexture;
+        private GUIStyle style;
+        private RectTransform cachedTransform; // The overhead of calling gameobject.transform every update is expensive
+
+        /// <summary>
+        /// User override.
+        /// </summary>
+        private bool shouldShowCursor;
+
+        /// <summary>
+        /// Uses the fact that <see cref="TextAnchor"/> is an enum and we can cast it to int.
+        /// This not only is more memory safe, and contingent which makes it faster, but also
+        /// allows us to iterate through it with ease which is called every frame to reset it,
+        /// and the cast overhead is minimal.
+        /// </summary>
+        private CursorTextBox[] textBoxes;
+
+        /// <summary>
+        /// Create a new mouse cursor with the supplied style.
+        /// </summary>
+        /// <param name="style"> If null it will instead use a default style which is 15pt Arial, alignment being middle center.</param>
+        public MouseCursor(GUIStyle style = null)
+        {
+            shouldShowCursor = true;
+            UIMode = false;
+
+            if (style == null)
+            {
+                this.style = new GUIStyle() { font = Resources.Load<Font>("Fonts/Arial/Arial") as Font, fontSize = 15, alignment = TextAnchor.MiddleCenter };
+            }
+
+            cursorTexture = Resources.Load<Texture2D>("UI/Cursors/Ship");
+            KeyboardManager.Instance.RegisterInputAction("ToggleCursorTextBox", KeyboardMappedInputType.KeyUp, () => { shouldShowCursor = !shouldShowCursor; });
```

Comment:
Here's where you still have the InputAction registered, it is useful if you can tie it directly to the coordinates (rather than as it currently is tied to the whole tooltip display), mainly because this is the sort of thing one might want to just toggle on for just a little bit to take a quick look at the coordinates, without having to go into settings.

Reply:
Oh I see, yeah that's easy to change.  My original point still stands I think though...?  I will change it however due to the simplicity of just toggling a different boolean.